src/equisolve/utils/split_data.py

```diff
@@ -0,0 +1,431 @@
+        ]
+    """
+    # Check input args and parse `tensors` and `names` into lists
+    tensors = [tensors] if isinstance(tensors, TensorMap) else tensors
+    names = [names] if isinstance(names, str) else names
+    _check_args(tensors, axis, names, n_groups, group_sizes, random_seed)
+
+    # Get array of unique indices to split by for each tensor in `tensors`
+    unique_idxs_list = [
+        equistore.unique_metadata(tensor, axis, names) for tensor in tensors
+    ]
+
+    # Check that the unique indices are equivalent for all input tensors
+    _check_labels_equivalent(unique_idxs_list)
+    unique_idxs = unique_idxs_list[0]
+
+    # Shuffle the unique indices according to the random seed
+    _shuffle_indices(unique_idxs, random_seed)
+
+    # Must be at least as many unique indices as groups
+    n_indices = len(unique_idxs)
+    if n_indices < n_groups:
+        raise ValueError(
+            f"the number of groups specified ({n_groups}) is greater than the"
+            + f" number of unique metadata indices ({n_indices}) for the"
+            + f" chosen axis {axis} and names {names}: {unique_idxs}"
+        )
+
+    # Get group sizes
+    group_sizes = _get_group_sizes(n_groups, len(unique_idxs), group_sizes)
+
+    # The sum of the absolute group sizes must be less than or equal to the
+    # number of unique indices
+    if n_indices < sum(group_sizes):
+        raise ValueError(
+            f"the sum of the absolute group sizes ({sum(group_sizes)}) is greater than "
+            + f"the number of unique metadata indices ({n_indices}) for the chosen "
+            + f"axis {axis} and names {names}: {unique_idxs}"
+        )
+
+    # Group the indices according to the group sizes
+    grouped_labels = _group_indices(unique_idxs, group_sizes)
+
+    # Split each of the input TensorMaps
+    split_tensors = []
+    for tensor in tensors:
+        split_tensors.append(equistore.split(tensor, axis, grouped_labels))
+
+    return split_tensors, grouped_labels
+
+
+def _shuffle_indices(indices: Labels, random_seed: Optional[int] = None):
+    """
+    Shuffles the input Labels object `indices` according to the random seed. If
+    `random_seed=None`, not shuffling is performed and the indices are just
+    returned. If -1, shuffling occurs but with no random seed set. If an integer
+    != -1, shuffling is executed but with a random seed set to this value. There
+    is no return value, as the input `indices` is shuffled in place.
+    """
+    if random_seed is not None:  # shuffle
+        if random_seed != -1:  # set a numpy random seed
+            np.random.seed(random_seed)
+        np.random.shuffle(indices)
+
+
+def _get_group_sizes(
+    n_groups: int,
+    n_indices: int,
+    group_sizes: Optional[Union[List[float], List[int]]] = None,
+) -> np.ndarray:
+    """
+    Parses the `group_sizes` arg from :py:func:`split_data` and returns an array
+    of group sizes in absolute numbers.
```

Comment:
I don't understand from the docstring what the function is doing. Sorry 😰.

Reply:
Added more explanation in the docstring